testsuite/tests/typing-poly/pr9603.ml

```diff
@@ -29,8 +29,5 @@ Error: This expression has type
          < m : 'left 'right. < left : 'left; right : 'right > pair >
        but an expression was expected of type
          < m : 'left 'right. < left : 'left; right : 'right > pair >
-       Type < left : 'left; right : 'right > pair = 'a * 'b
-       is not compatible with type < left : 'left0; right : 'right0 > pair
-       The method left has type 'a, but the expected method type was 'left
-       The universal variable 'left would escape its scope
+       Types for method m are incompatible
```

Comment:
This seems like a regression. There is no information left in the remaining error message
whereas the sibylline preexisting message at least points to an issue with universal type variables.

Reply:
I basically disagree. The old error message is kind of nonsense, caused by one unification error leaking and being fused with another one. The new error message at least only says things that make sense.

Of course neither error message gives a real user a hope in hell of finding the actual issue, but that is because this is basically a bug. We could try to give a specific error message for this issue but, since @trefis is currently working on some changes that will fix it, I don't think it is worth the effort.

Reply:
Fair enough. I agree that both error messages are basically undecipherable without some internal knowledge of the typechecker.

Reply:
If we're all agreed, then shall we mark this as resolved?